### ISSUE TITLE
headerCreate is slow

### ISSUE THREAD

**Issue by eatkins** [akka/akka#27604]
I was testing sbt performance in the akka project and noticed that if I ran `Test / definedTests` in the root project after I had fully compiled everything, it took 25 seconds to return. 15 of those seconds were spent creating headers across the various projects. Given how relatively infrequently new files are added or the header itself is updated, I'd suggest removing:
```
  def additional: Seq[Def.Setting[_]] = Def.settings(
    (compile in Compile) := {
      (headerCreate in Compile).value                                          
      (compile in Compile).value                                 
    },                    
    (compile in Test) := {                                                                            
      (headerCreate in Test).value                             
      (compile in Test).value
    }                                  
  )
```
from the build.sbt and running headerCheck during ci. That is what sbt does.

**Comment by patriknw** [comment]
Ok, it would indeed be good to improve that. Would it be possible to improve the plugin itself? E.g. does it make use of the sbt caching for touched files?

This is how the top timing looks like for me:
```
Total time: 9685 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-bench-jmh-typed") / update                                      : 235 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-bench-jmh") / update                                            : 194 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-docs") / update                                                 : 146 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-cluster-metrics") / update                                      : 143 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-cluster-sharding-typed") / update                               : 139 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-stream") / Compile / headerCreate                               : 138 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-cluster-typed") / update                                        : 138 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-remote") / Compile / headerCreate                               : 135 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-cluster-sharding") / update                                     : 134 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-distributed-data") / update                                     : 120 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-distributed-data") / Compile / headerCreate                     : 115 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-remote-tests") / update                                         : 108 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-cluster-tools") / update                                        : 108 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-protobuf") / Compile / headerCreate                             : 107 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-cluster") / update                                              : 104 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-actor-typed-tests") / Test / headerCreate                       : 104 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-actor") / Compile / headerCreate                                :  98 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-remote") / update                                               :  94 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-stream-tests") / Test / headerCreate                            :  91 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-docs") / Test / headerCreate                                    :  87 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-persistence-typed") / update                                    :  87 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-stream-tests") / update                                         :  81 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-coordination") / update                                         :  80 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-serialization-jackson") / update                                :  79 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-docs") / Test / compileIncremental                              :  72 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-cluster") / Compile / headerCreate                              :  63 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-actor-tests") / Test / headerCreate                             :  58 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-actor-tests") / update                                          :  58 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-discovery") / update                                            :  57 ms
  ProjectRef(uri("file:/Users/patrik/dev/akka/"), "akka-protobuf-v3") / assembly                                        :  57 ms
```

**Comment by johanandren** [comment]
Caching PR in progress https://github.com/sbt/sbt-header/pull/170

**Comment by jrudolph** [comment]
Fixed in sbt-header 5.3.0. Thanks a lot @johanandren, that helps a lot!
